### PR TITLE
[INLONG-8039][Manager] Optimize the transform interface to support pagination query

### LINKED ISSUE #8039: [Improve][Manager] Optimize the transform interface
### Description

1. Add an interface to get a single transform.
2. Add pagination to the query interface.
...

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by haifxu**
### Prepare a Pull Request

- Fixes #8039 

### Motivation

Support pagination query transform

### Modifications

1. Add an interface to get a single transform.
2. Add pagination to the query interface.
3. Modify `GET` to `POST` for the list interface.

### Verifying this change

- [x] This change is already covered by existing tests, such as:
manager/service/transform/StreamTransformServiceTest.java

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformServiceImpl.java [hunk 1]
```diff
@@ -139,16 +142,35 @@ public Integer save(TransformRequest request, UserInfo opInfo) {
...
+        String streamId = request.getInlongStreamId();
         LOGGER.debug("begin to fetch transform info by groupId={} and streamId={} ", groupId, streamId);
-        Preconditions.expectNotBlank(groupId, ErrorCodeEnum.GROUP_ID_IS_EMPTY);
-        List<StreamTransformEntity> entityList = transformMapper.selectByRelatedId(groupId, streamId, null);
-        if (CollectionUtils.isEmpty(entityList)) {
-            return Collections.emptyList();
-        }
+        PageHelper.startPage(request.getPageNum(), request.getPageSize());
+        PageResult<TransformResponse> pageResponse = new PageResult<>();
+        pageResponse.setList(getTransform(groupId, streamId));
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformServiceImpl.java:151]: For the scalability, it's better to make a **selectByCondition** method in **StreamTransformMapper** which accept **TransformPageRequest** and select entities by those condition defined in page request.

**Review by vernedeng — CHANGES_REQUESTED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformService.java [hunk 2]
```diff
@@ -47,14 +49,31 @@ public interface StreamTransformService {
...
+    TransformResponse get(Integer id, UserInfo opInfo);
+
     /**
      * Query transform information based on inlong group id and inlong stream id.
      *
      * @param groupId the inlong group id
      * @param streamId the inlong stream id
      * @return the transform response
      */
-    List<TransformResponse> listTransform(String groupId, String streamId);
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformService.java:57]: No need to change this method name and suggest changing the Page Request to `listByCondition` or `listByPagination`.

If you change to `listByPagination`, suggest changing the other interfaces alse, including StreamSource, and StreamSink.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformServiceImpl.java (hunk 1 above)
Comment (haifxu) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformServiceImpl.java]: Done.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformService.java (hunk 2 above)
Comment (haifxu) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformService.java]: Done.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformServiceImpl.java [hunk 3]
```diff
@@ -139,16 +142,37 @@ public Integer save(TransformRequest request, UserInfo opInfo) {
...
+        return pageResponse;
+    }
+
+    @Override
+    public TransformResponse get(Integer id, UserInfo opInfo) {
+        StreamTransformEntity entity = transformMapper.selectById(id);
+        List<StreamTransformFieldEntity> fieldEntities = transformFieldMapper.selectByTransformId(id);
+        if (entity == null) {
+            throw new BusinessException(ErrorCodeEnum.TRANSFORM_NOT_FOUND,
+                    String.format("source not found by id=%s", id));
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformServiceImpl.java:171]: transform not found

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformServiceImpl.java [hunk 4]
```diff
@@ -357,6 +352,45 @@ public Boolean delete(DeleteTransformRequest request, UserInfo opInfo) {
...
+            int transformId = transformResponse.getId();
+            List<StreamField> fieldInfos = fieldInfoMap.get(transformId);
+            if (CollectionUtils.isNotEmpty(fieldInfos)) {
+                transformResponse.setFieldList(fieldInfos);
+            }
+        });
+        return transformResponses;
+    }
+
+    private void checkUser(String groupId, UserInfo userInfo) {
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/transform/StreamTransformServiceImpl.java:379]: please use userService.checkUser

**Review by vernedeng — APPROVED** [review]

**Review by healchow — APPROVED** [review]
LGTM

**Review by fuweng11 — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**